migrations/2020_06_27_000000_set_default_theme.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of fof/nightmode.
+ *
+ * Copyright (c) 2020 FriendsOfFlarum.
+ *
+ *  For the full copyright and license information, please view the LICENSE.md
+ *  file that was distributed with this source code.
+ *
+ */
+
+use Illuminate\Database\Schema\Builder;
+
+return [
+    'up' => function (Builder $schema) {
+        /**
+         * @var \Flarum\Settings\SettingsRepositoryInterface
+         */
+        $settings = app('flarum.settings');
+
+        $settings->set('fof-nightmode.default_theme', '0');
+    },
+    'down' => function (Builder $schema) {
+        $settings = app('flarum.settings');
+
+        $settings->delete('fof-nightmode.default_theme');
+    },
+];
```

Comment:
The `Builder` class isn't really being utilized in this code at all as well as the `$schema` parameter.

@imorland gave you a starting point, but may I suggest taking a look at [this example](https://github.com/flarum/tags/blob/master/migrations/2015_02_24_000000_set_default_settings.php) instead? It handles both up and down functions for this migration and you can even look at how the shortcut works [here](https://github.com/flarum/core/blob/71e313e677402fca76ffed0ffe5863bb7096aaff/src/Database/Migration.php#L121-L146) as well.

Reply:
That is a lot cleaner!

I need to sit down and look through all the Flarum API docs soon. The official docs seem quite outdated and don't go very in-depth at all, sadly.

I never would have found this!

Reply:
@davwheat no worries, we are aware of the situation with the official docs but rest assured, we are looking into fixing it up :)

Keep up the good work, this PR is in better shape now and we all learn from this peer-review experience :D

Reply:
The best way to find what you want is to just look at other extensions on a deeper level to see how they use Flarum's core classes and components and how they extend on them or how they are used.

There is also [unofficial API docs](https://api.flarum.dev/) you can take look at but if you really want to understand how Flarum works, just take a look at the [core classes](https://github.com/flarum/core/tree/71e313e677402fca76ffed0ffe5863bb7096aaff/src) :)

Reply:
Lastly, I want also to point out that even though I did point out in my first comment that `$settings->set('fof-nightmode.default_theme', '0');` was set as a string and not an integer, with the shortcut migration, is okay to add single quotes because of how the shortcut works. Seeing as it [creates an instance](https://github.com/flarum/core/blob/71e313e677402fca76ffed0ffe5863bb7096aaff/src/Database/Migration.php#L128) of the `DatabaseSettingsRepository` class within the `addSettings()` method, single quotations are okay to use since it's based on key-value pairs.

Now, a more seasoned developer could most likely explain that better but it's how I understand it mostly haha.